### PR TITLE
Add Gravatar Support

### LINKED ISSUE #640: [F.Rq.] Gravatar
tl;dr: allows users to change their avatar on many sites at once, and auto-generates unique avatars when a user doesn't provide one. You've probably seen the latter e.g. on GitHub. [API docs](https://en.gravatar.com/site/implement)

update 2024-12: TIL there's an open alternative to Gravatar called [ivatar](https://wiki.libravatar.org/features/). It says it can fall back to Gravatar so maybe it's a direct upgrade?

see also #227

### PR DESCRIPTION
**Pull request by adelikat**
This PR fixes a few things on Profile/Settings:
Show an "Image too big" message when image is too big (it was already making some things red and preventing save, but this helps make it more clear"
Fix validation messages to be red
Removed the preview button for avatars and have it automatically preview as you type/edit the avatar box

In addition, it adds a toggle for "Use Gravatar" that will automatically generate the gravatar url based on the user's email, with an option to use a different email for the gravatar compared to their tasvideos account email

When Gravatar is not used:
image

When activating gravatar:
image

As you type in the Gravatar email box, it will automatically update the avatar box (which will automatically update the preview)

resolves #640

### CODE REVIEW

File: TASVideos/Pages/Profile/Settings.cshtml [hunk 1]
```diff
@@ -148,45 +163,86 @@
...
 				avatarImgElem.src = avatar;
 			} else {
-				previewElem.classList.add('d-none');
+				avatarImgElem.src = '';
 				preventSave(false);
 			}
 		}
 
 		function validateAvatar() {
+			console.log('validating avatar requirements');
```
Comment (Masterjun3) [inline TASVideos/Pages/Profile/Settings.cshtml:187]: We probably don't want stray console.log's.

File: TASVideos/Pages/Profile/Settings.cshtml [hunk 2]
```diff
@@ -148,45 +163,86 @@
...
+			const checked = document.querySelector('[name="UseGravatar"]:checked').value == 'True';
+			if (checked) {
+				console.log('showing gravatar')
+				document.getElementById('gravatar-section').classList.remove('d-none');
+				await generateGravatarPreview();
+			} else {
+				console.log('showing avatar')
+				document.getElementById('gravatar-section').classList.add('d-none');
+			}
+			console.log('checked', checked)
```
Comment (Masterjun3) [inline TASVideos/Pages/Profile/Settings.cshtml:220]: We probably don't want stray console.log's.

File: TASVideos/Pages/Profile/Settings.cshtml [hunk 3]
```diff
@@ -148,45 +163,86 @@
...
 			const maxWidth = @AvatarRequirements.Width;
 			const maxHeight = @AvatarRequirements.Height;
+			const descSection = document.getElementById('avatar-description');
 
 			const tooBig = avatarImgElem.width > maxWidth || avatarImgElem.height > maxHeight;
 			if (tooBig) {
-				previewElem.classList.add('alert');
-				previewElem.classList.add('alert-danger');
+				descSection.classList.add('alert');
+				descSection.classList.add('alert-danger');
```
Comment (Masterjun3) [inline TASVideos/Pages/Profile/Settings.cshtml:195]: Setting an `alert` on a `<small>` element breaks the layout.
image

File: TASVideos/Pages/Profile/Settings.cshtml (hunk 1 above)
Comment (adelikat) [reply TASVideos/Pages/Profile/Settings.cshtml]: Yup, that was an accident, fixed

**Review by YoshiRulz — COMMENTED** [review]
I can't test locally but the interface LGTM.

**Comment by YoshiRulz** [comment]
When I change the value in "Email to Use for Gravatar Image", the URI field and image preview both change, but upon saving they're all reset to my account email.

**Pull request merged**